src/web/wms/src/main/java/org/geoserver/wms/web/data/AbstractStylePage.java

```diff
@@ -306,8 +306,12 @@ protected void onError(AjaxRequestTarget target, Form<?> form) {
         });
         add(new AjaxSubmitLink("submit", styleForm) {
             @Override
-            protected void onSubmit(AjaxRequestTarget target, Form<?> form) {
-                doReturn(StylePage.class);
+            protected void onAfterSubmit(AjaxRequestTarget target, Form<?> form) {
+                if (form.hasError()) {
+                    target.add(feedbackPanel);
+                } else {
+                    doReturn(StylePage.class);
+                }
```

Comment:
This is a fairly important workflow fix, kind of sorry it missed the 2.10.x release :(